### PR TITLE
feat(rest): Create new endpoint to cancel all scheduled services.

### LINKED ISSUE #2237: feat(rest): Create new endpoint to cancel all scheduled services in schedule tab.
[//]: # (Copyright Siemens AG, 2021. Part of the SW360 Portal Project)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

### Description

Create new endpoint to cancel all the schedule task. means if any numbers of schedule services are running and we can cancel all the task on single service call. 

Note: Users should have admin access.

#### Steps followed and expected result

Describe the steps followed by you and your expected results after following the steps.

### PR DESCRIPTION
**Pull request by nikkuma7**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Closes #2237 

### How To Test?

http://localhost:8080/resource/api/schedule/unscheduleAllService

> How should these changes be tested by the reviewer?
> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by keerthi-bl** [comment]
Test completed.

2238_issue_ref_screenshot

File: rest/resource-server/src/docs/asciidoc/schedule.adoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// SPDX-License-Identifier: EPL-2.0
+//
+
+[[resources-schedule]]
+=== Schedule
+
+The Schedule resource is used to get and list the Schedule requests.
+
+[[unschedule-services]]
+==== cancle all schedule.
```
Comment (rudra-superrr) [inline rest/resource-server/src/docs/asciidoc/schedule.adoc:17]: cancle -> Cancel
correct this at all places where cancle is used.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/schedule/ScheduleAdminController.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    private Sw360ScheduleService scheduleService;
+    
+
+    @Override
+    public RepositoryLinksResource process(RepositoryLinksResource resource) {
+        resource.add(linkTo(ScheduleAdminController.class).slash("api/schedule").withRel("schedule"));
+        return resource;
+    }
+
+    @RequestMapping(value = SCHEDULE_URL + "/unscheduleAllService", method = RequestMethod.DELETE)
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/schedule/ScheduleAdminController.java:54]: /unscheduleAllService -> /unscheduleAllServices

File: rest/resource-server/src/docs/asciidoc/schedule.adoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+// SPDX-License-Identifier: EPL-2.0
+//
+
+[[resources-schedule]]
+=== Schedule
+
+The Schedule resource is used to get and list the Schedule requests.
+
+[[unschedule-services]]
+==== cancel all schedule.
```
Comment (rudra-superrr) [inline rest/resource-server/src/docs/asciidoc/schedule.adoc:17]: replace it with Cancel all scheduled services.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/schedule/ScheduleAdminController.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    @Override
+    public RepositoryLinksResource process(RepositoryLinksResource resource) {
+        resource.add(linkTo(ScheduleAdminController.class).slash("api/schedule").withRel("schedule"));
+        return resource;
+    }
+
+    @RequestMapping(value = SCHEDULE_URL + "/unscheduleAllServices", method = RequestMethod.DELETE)
+    public ResponseEntity<?> unscheduleAllServices()throws TException {
+        User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        RequestStatus requestStatus = scheduleService.cancleAllServices(sw360User);
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/schedule/ScheduleAdminController.java:57]: change service name to cancelAllServices from cancleAllServices.

Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/schedule/ScheduleAdminController.java]: done.

File: rest/resource-server/src/docs/asciidoc/schedule.adoc (hunk 3 above)
Comment (nikkuma7) [reply rest/resource-server/src/docs/asciidoc/schedule.adoc]: done.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/schedule/ScheduleAdminController.java (hunk 2 above)
Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/schedule/ScheduleAdminController.java]: done.

**Comment by rudra-superrr** [comment]
code looks good.

**Pull request merged**